modules/account-lib/src/coin/algo/transaction.ts

```diff
@@ -210,14 +206,22 @@ export class Transaction extends BaseTransaction {
       tokenId: this._algoTransaction?.assetIndex,
       genesisID: this._algoTransaction.genesisID,
       genesisHash: this._algoTransaction.genesisHash.toString('base64'),
+      txType: TransactionType[this.type],
+      amount: '0',
     };
+
     if (this._algoTransaction.closeRemainderTo) {
       result.closeRemainderTo = algosdk.encodeAddress(this._algoTransaction.closeRemainderTo.publicKey);
     }
-    if (this.type === TransactionType.Send) {
+
+    if (this._algoTransaction.to) {
       result.to = algosdk.encodeAddress(this._algoTransaction.to.publicKey);
+    }
+
+    if (this._algoTransaction.amount) {
       result.amount = this._algoTransaction.amount.toString();
     }
```

Comment:
you could do something like this instead the if statement, if you like.
result.amount = this._algoTransaction.amount?.toString();

Reply:
Your expression will set the undefined value to "result.amount". The 'if' will not add the field to 'result' structure at all. They similar but not totally equivalent. I'm unsure on the better way in this case.